openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +388,226 @@ def debug_cohort_mgmt(request, course_key_string):
         kwargs={'course_key': text_type(course_key)}
     )}
     return render_to_response('/course_groups/debug.html', context)
+
+
+class APIPermissions(APIView):
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
+
+
+class CohortSettings(DeveloperErrorViewMixin, APIPermissions):
+    """
+    **Use Cases**
+
+        Get the cohort setting for a course.
+        Set the cohort setting for a course.
+
+    **Example Requests**:
+
+        GET /api/cohorts/v1/settings/{course_id}
+        PUT /api/cohorts/v1/settings/{course_id}
+
+    **Response Values**
+
+        * is_cohorted: current status of the cohort setting
+    """
+
+    def get(self, request, course_key_string):
+        """
+        Endpoint to fetch the course cohort settings.
+        """
+        course_key, _ = api.get_course(request, course_key_string)
+        return Response(_cohort_settings(course_key))
+
+    def put(self, request, course_key_string):
+        """
+        Endpoint to set the course cohort settings.
+        """
+        course_key, _ = api.get_course(request, course_key_string)
+
+        if 'is_cohorted' not in request.data:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Missing field "is_cohorted".')
+        try:
+            cohorts.set_course_cohorted(course_key, request.data.get('is_cohorted'))
+        except ValueError as err:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST, err)
+        return Response(_cohort_settings(course_key))
+
+
+class CohortHandler(DeveloperErrorViewMixin, APIPermissions):
+    """
+    **Use Cases**
+
+        Get the current cohorts in a course.
+        Create a new cohort in a course.
+        Modify a cohort in a course.
+
+    **Example Requests**:
+
+        GET /api/cohorts/v1/courses/{course_id}/cohorts
+        POST /api/cohorts/v1/courses/{course_id}/cohorts
+        GET /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}
+        PATCH /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}
+
+    **Response Values**
+
+        * cohorts: List of cohorts.
+        * cohort: A cohort representation:
+            * name: The string identifier for a cohort.
+            * id: The integer identifier for a cohort.
+            * user_count: The number of students in the cohort.
+            * assignment_type: The string representing the assignment type.
+            * user_partition_id: The integer identified of the UserPartition.
+            * group_id: The integer identified of the specific group in the partition.
+    """
+
+    def get(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to get either one or all cohorts.
+        """
+        course_key, course = api.get_course(request, course_key_string)
```

Comment:
Originally this was using `SlashSeparatedCourseKey.from_deprecated_string` but now it's using `CourseKey.from_string`. I'm not familair with the history here so not sure if this can cause issues, but perhaps it's best to not change this behaviour.

Reply:
Ok 👍

Reply:
I guess it's for the old course ids of the form org/xyz/2018. I'm not sure TBH.

Reply:
Ok, updated the code to use it.

Reply:
Yeah, `SlashSeparatedCourseKey` is for the old style of Course IDs. The code for that is in the opaque-keys repo, but you should pretty much always use `CourseKey.from_string` which knows how to parse all kinds of course IDs and will create the appropriate subclass as needed.

Reply:
@ormsbee Thanks for that info!